### PR TITLE
Depl-140848 Store script execution output to db_script table

### PR DESCRIPTION
**Pull request by bakirstagram**
Depl-140848-115167 [DEV] Store script execution output to db_script table

### CODE REVIEW

**Review by Simplycissimys — CHANGES_REQUESTED** [review]

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 1]
```diff
@@ -76,7 +76,17 @@ public int execute(SqlScript script) {
                                            script.getFile().getAbsolutePath()));
 
             Instant start = Instant.now();
+
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            PrintStream ps = new PrintStream(baos);
+            PrintStream old = System.out;
+            System.setOut(ps);
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: fixed

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 2]
```diff
@@ -71,28 +71,29 @@ public int execute(SqlScript script) {
...
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            PrintStream ps = new PrintStream(baos);
-            PrintStream old = System.out;
-            System.setOut(ps);
+            executor.setOut(ctx.getOutputStream());
             executor.run();
-            System.out.flush();
-            System.setOut(old);
             script.setOutput(baos.toString());
             System.out.println(baos.toString());
```
Comment (Simplycissimys) [inline src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java:85]: the script output should be written to the console as available, not when script execution is finished, find a way to make `executor` write into 2 streams: `system.out` and stream for saving into db

**Review by Simplycissimys — CHANGES_REQUESTED** [review]

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 3]
```diff
@@ -71,28 +71,29 @@ public int execute(SqlScript script) {
...
+            bos.flush();
+            bos.close();
+            baos.close();
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 4]
```diff
@@ -71,28 +71,29 @@ public int execute(SqlScript script) {
...
-                                           script.getFile().getAbsolutePath()));
+                    script.getFile().getAbsolutePath()));
 
             Instant start = Instant.now();
-
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            PrintStream ps = new PrintStream(baos);
-            PrintStream old = System.out;
-            System.setOut(ps);
+            executor.setOut(ctx.getOutputStream());
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 5]
```diff
@@ -71,28 +71,29 @@ public int execute(SqlScript script) {
...
+            baos.close();
+
 
             String scriptExecutionTime = formatDurationHMS(Duration.between(start, Instant.now()).toMillis());
 
             logger.info("\n[{}] runtime: {}", GREEN, script.getName(), scriptExecutionTime);
 
             return (boolean) ctx.getProperty(ERR_ENCOUNTERED) ? SCRIPT_EXIT_CODE_ERROR : SCRIPT_EXIT_CODE_SUCCESS;
-        } catch (SQLException e) {
+        } catch (SQLException | IOException e) {
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java (hunk 2 above)
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 6]
```diff
@@ -60,23 +61,32 @@ public class SqlScriptExecutor {
...
+                cnnCredentials.getSchemaWithUrlBeforeDot(), ZonedDateTime.now().format(ISO_TIME));
+        script.setOutput("execute");
 
         File workingDir = script.getFile().getParentFile();
         File wrapperScriptFile = getTmpWrapperScript(script.getSchemaType(), workingDir);
 
-        try (Connection connection = getConnection(script.getSchemaType(), cnnCredentials.getSchemaName())) {
+        try (Connection connection = getConnection(script.getSchemaType(), cnnCredentials.getSchemaName());
+             ByteArrayOutputStream baos = new ByteArrayOutputStream();
+             BufferedOutputStream buff = new BufferedOutputStream(baos)) {
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

File: src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java [hunk 7]
```diff
@@ -85,6 +95,9 @@ public int execute(SqlScript script) {
         } catch (SQLException e) {
             logger.error("Error during connection DB.", e);
             return SCRIPT_EXIT_CODE_ERROR;
+        } catch (IOException e) {
+            logger.error("I/O Error.", e);
```
[the remark this answers is not in the document]
Comment (bakirstagram) [reply src/main/java/com/onevizion/scmdb/SqlScriptExecutor.java]: done

**Review by Simplycissimys — APPROVED** [review]

**Pull request merged**